### PR TITLE
Resolve doc build issues (#162, #163)

### PR DESCRIPTION
**Pull request by hanbyul-kim**
Hi!

1. Fixing #162 
There have been 11 warnings while building docs. They were mostly indentation errors. This may cause future issues, I resolved them at 29c2adf182c1bb7e9a03bcd4f5909fde7889e34c. Currently 0 warnings.

2. Fixing #163 
https://sonnet.readthedocs.io/en/latest/api.html#modes
https://sonnet.readthedocs.io/en/latest/api.html#scope
code examples weren't rendered properly before. Fixed here:  0e8d1c40ec212b83610f3dacdc9d4f809eb559d7

### CODE REVIEW

File: sonnet/src/mixed_precision.py [hunk 1]
```diff
@@ -77,21 +77,23 @@ def _cast_call(f, new_dtype, args, kwargs):
 def modes(valid_types):
   """Decorate a function to cast inputs/outputs to different precision.
 
-      snt.Linear.__call__ = snt.mixed_precision.modes(
-        [tf.float32, tf.float16])(snt.Linear.__call__)
-      mod = snt.Linear(10)
-      snt.mixed_precision.enable(tf.float16)
-      y = mod(tf.ones([1, 1]))  # First call will be done in F32.
-      y = mod(tf.ones([1, 1]))  # MatMul/Add will be done in F16.
+      .. code-block:: python
```
Comment (superbobry) [inline sonnet/src/mixed_precision.py]: Would you mind using `::` instead as in e.g. 

https://github.com/deepmind/sonnet/blob/3732f195416d8d0e212835beb65303abfab8929f/sonnet/src/recurrent.py#L226

File: sonnet/src/mixed_precision.py [hunk 2]
```diff
@@ -77,21 +77,23 @@ def _cast_call(f, new_dtype, args, kwargs):
...
   Args:
     valid_types: Collection of types that the function being decorated is legal
-      to run in.
+    to run in.
 
   Returns:
     A decorator that will cast the inputs and outputs of the decorated function
     according to the global mixed precision policy and the functions eligibility
     for mixed precision.
+
```
Comment (superbobry) [inline sonnet/src/mixed_precision.py]: This is an extra newline, right? If so, could you remove it please? Otherwise it might fail our internal linter checks.

File: sonnet/src/mixed_precision.py [hunk 3]
```diff
@@ -146,14 +148,16 @@ def scope(dtype: tf.DType):
...
 
-      snt.mixed_precision.enable(tf.float32)
-      snt.Linear.__call__ = snt.mixed_precision.modes(
-        [tf.float32, tf.float16])(snt.Linear.__call__)
-      mod = snt.Linear(10)
-      with snt.mixed_precision.scope(tf.float16):
-        y = mod(tf.ones([1, 1]))  # First call will be done in F32.
-        y = mod(tf.ones([1, 1]))  # MatMul/Add will be done in F16.
-      y = mod(tf.ones([1, 1]))  # Outside the scope will be done in F32.
+  .. code-block:: python
```
Comment (superbobry) [inline sonnet/src/mixed_precision.py]: Same as above, please use `::`.

File: sonnet/src/nets/vqvae.py [hunk 4]
```diff
@@ -94,9 +94,10 @@ def __call__(self, inputs, is_training):
         loss: Tensor containing the loss to optimize.
         perplexity: Tensor containing the perplexity of the encodings.
         encodings: Tensor containing the discrete encodings, ie which element
-          of the quantized space each input element was mapped to.
+        of the quantized space each input element was mapped to.
         encoding_indices: Tensor containing the discrete encoding indices, ie
-          which element of the quantized space each input element was mapped to.
+        which element of the quantized space each input element was mapped to.
+
```
Comment (superbobry) [inline sonnet/src/nets/vqvae.py]: Remove extra newline.

File: sonnet/src/nets/vqvae.py [hunk 5]
```diff
@@ -238,9 +239,10 @@ def __call__(self, inputs, is_training):
...
```
Comment (superbobry) [inline sonnet/src/nets/vqvae.py]: Remove extra newline.

File: sonnet/src/optimizers/momentum.py [hunk 6]
```diff
@@ -87,8 +87,7 @@ def apply(self, updates: Sequence[types.ParameterUpdate],
     And when using Nesterov momentum (`use_nesterov=True`) it applies:
 
         accum_t <- momentum * accum_{t-1} + update
-        parameter <- parameter - (learning_rate * update +
-                                  learning_rate * momentum * accum_t)
+        parameter <- parameter - (learning_rate * update + learning_rate * momentum * accum_t)
```
Comment (superbobry) [inline sonnet/src/optimizers/momentum.py]: OOC, is this change necessary?

**Review by superbobry — CHANGES_REQUESTED** [review]
Hey @hanbyul-kim, thanks for fixing the docs! I've left a few comments.

File: sonnet/src/nets/vqvae.py (hunk 4 above)
Comment (hanbyul-kim) [reply sonnet/src/nets/vqvae.py]: Resolved. e7f7b61

File: sonnet/src/nets/vqvae.py (hunk 5 above)
Comment (hanbyul-kim) [reply sonnet/src/nets/vqvae.py]: Resolved. e7f7b61

File: sonnet/src/mixed_precision.py (hunk 1 above)
Comment (hanbyul-kim) [reply sonnet/src/mixed_precision.py]: Resolved. 623a88d

File: sonnet/src/mixed_precision.py (hunk 3 above)
Comment (hanbyul-kim) [reply sonnet/src/mixed_precision.py]: Resolved. 623a88d

File: sonnet/src/optimizers/momentum.py (hunk 6 above)
Comment (hanbyul-kim) [reply sonnet/src/optimizers/momentum.py]: Those lines showed unexpected indentation warning. If you don't like it, how about this? 
8574df2

**Comment by hanbyul-kim** [comment]
> Hey @hanbyul-kim, thanks for fixing the docs! I've left a few comments.

I resolved all the comments you gave me last night and rebased v2 again. Please review them again.

File: sonnet/src/mixed_precision.py [hunk 7]
```diff
@@ -144,16 +143,16 @@ def _wrapper(f, instance, args, kwargs):
...
 
-      snt.mixed_precision.enable(tf.float32)
-      snt.Linear.__call__ = snt.mixed_precision.modes(
-        [tf.float32, tf.float16])(snt.Linear.__call__)
-      mod = snt.Linear(10)
-      with snt.mixed_precision.scope(tf.float16):
-        y = mod(tf.ones([1, 1]))  # First call will be done in F32.
-        y = mod(tf.ones([1, 1]))  # MatMul/Add will be done in F16.
-      y = mod(tf.ones([1, 1]))  # Outside the scope will be done in F32.
+       snt.mixed_precision.enable(tf.float32)
```
Comment (superbobry) [inline sonnet/src/mixed_precision.py]: Just a nit: this is 5 spaces instead of 4.

**Review by superbobry — APPROVED** [review]
Thanks for addressing the comments!

File: sonnet/src/mixed_precision.py (hunk 7 above)
Comment (hanbyul-kim) [reply sonnet/src/mixed_precision.py]: My mistake. Resolved. c8fbd10

**Review by superbobry — APPROVED** [review]

**Review by tomhennigan — APPROVED** [review]
